lib/bookmark_manager.rb

```diff
@@ -136,20 +138,30 @@ def update_topic_user_bookmarked(topic, opts = {})
     # PostCreator can specify whether auto_track is enabled or not, don't want to
     # create a TopicUser in that case
     return if opts.key?(:auto_track) && !opts[:auto_track]
-    TopicUser.change(@user.id, topic, bookmarked: Bookmark.for_user_in_topic(@user.id, topic.id).exists?)
+    TopicUser.change(
+      @user.id,
+      topic,
+      bookmarked: Bookmark.for_user_in_topic(@user.id, topic.id).exists?,
+    )
   end
 
   def bookmark_model_options_with_defaults(options)
-    model_options = {
-      pinned: options[:pinned]
-    }
+    model_options = { pinned: options[:pinned] }
 
     if options[:auto_delete_preference].blank?
-      model_options[:auto_delete_preference] = Bookmark.auto_delete_preferences[:never]
+      model_options[:auto_delete_preference] = if user_auto_delete_preference.present?
```

Comment:
Not sure if my personal preference, but I think this is a bit more clear:

```ruby
model_options[:auto_delete_preference] = 
  options[:auto_delete_preference].presence ||
  @guardian.user.user_option&.bookmark_auto_delete_preference.presence ||
  Bookmark.auto_delete_preferences[:clear_reminder]
```

Reply:
I don't find that at all more readable personally :) Anyway, I've run this through `syntax_tree` and this is how it formatted it, so leaving as is